src/messages/proposals.rs

```diff
@@ -233,17 +236,33 @@ impl<'a> ProposalQueue<'a> {
     pub(crate) fn from_committed_proposals(
         ciphersuite: &Ciphersuite,
         committed_proposals: &'a [ProposalOrRef],
-        proposals_by_reference: &'a ProposalQueue<'a>,
+        proposals_by_reference: &[&'a MLSPlaintext],
         sender: Sender,
     ) -> Result<Self, ProposalQueueError> {
+        // Feed the `proposals_by_reference` in a `HashMap` so that we can easily
+        // extract then by reference later
+        let mut proposals_by_reference_queue: HashMap<ProposalReference, QueuedProposal> =
+            HashMap::new();
+        for mls_plaintext in proposals_by_reference {
+            let queued_proposal =
+                QueuedProposal::from_mls_plaintext(ciphersuite, mls_plaintext).unwrap();
+            proposals_by_reference_queue.insert(
+                queued_proposal.proposal_reference().clone(),
```

Comment:
Wouldn't it make more sense to put the reference in here?

Reply:
I'm not sure what you mean. We do put the reference in a HashMap there. Where would you suggest we put it instead?